bin/i18n/resources/dashboard/course_content/sync_in.rb

```diff
@@ -0,0 +1,415 @@
+                  name = behavior.at_xpath('./title[@name="NAME"]')
+                  description = behavior.at_xpath('./mutation/description')
+                  i18n_strings['behavior_names'][name.content] = name.content if name
+                  i18n_strings['behavior_descriptions'][description.content] = description.content if description
+                end
+
+                ## Variable Names
+                variables_get = blocks.xpath("//block[@type=\"variables_get\"]")
+                variables_set = blocks.xpath("//block[@type=\"variables_set\"]")
+                variables = variables_get + variables_set
+                i18n_strings[VARIABLE_NAMES_TYPE] = Hash.new unless variables.empty?
+                variables.each do |variable|
+                  name = variable.at_xpath('./title[@name="VAR"]')
+                  i18n_strings[VARIABLE_NAMES_TYPE][name.content] = name.content if name
+                end
+
+                ## Parameter Names
+                parameters = blocks.xpath("//block[@type=\"parameters_get\"]")
+                i18n_strings[PARAMETER_NAMES_TYPE] = Hash.new unless parameters.empty?
+                parameters.each do |parameter|
+                  name = parameter.at_xpath('./title[@name="VAR"]')
+                  i18n_strings[PARAMETER_NAMES_TYPE][name.content] = name.content if name
+                end
+
+                ## Placeholder texts
+                i18n_strings['placeholder_texts'] = i18n_strings['placeholder_texts'] || Hash.new
+                i18n_strings['placeholder_texts'].merge! get_all_placeholder_text_types(blocks)
+              end
+            end
+
+            if level.is_a? BubbleChoice
+              i18n_strings['sublevels'] = {}
+              # Block categories, variables, and parameters are handled differently below and are generally covered by the script levels
+              ignored_types = %w[block_categories variable_names parameter_names]
+              level.sublevels.map do |sublevel|
+                i18n_strings['sublevels'][sublevel.name] = get_i18n_strings(sublevel)
+
+                ignored_types.each do |type|
+                  i18n_strings['sublevels'][sublevel.name].delete(type) if i18n_strings['sublevels'][sublevel.name].key?(type)
+                end
+              end
+            end
+
+            if level.is_a? LevelGroup
+              i18n_strings['sublevels'] = {}
+              level.child_levels.map do |sublevel|
+                i18n_strings['sublevels'][sublevel.name] = get_i18n_strings sublevel
+              end
+            end
+
+            i18n_strings['contained levels'] = level.contained_levels.map do |contained_level|
+              get_i18n_strings(contained_level)
+            end
+
+            i18n_strings.delete_if {|_, value| value.blank?}
+          end
+
+          def write_to_yml(type, strings)
+            File.open(CDO.dir(File.join(I18N_SOURCE_DIR, "dashboard/#{type}.yml")), 'w') do |file|
```

Comment:
Could we add a `File.mkdir_p` line here? 
The sync fails if running from scratch, i.e if the `i18n/locales/source/dashboard` directory did not exists before.